### ISSUE TITLE
Change default work directory

### ISSUE THREAD

**Issue by PekkaSavolainen** [Spine-project/Spine-Toolbox#1462]
Change default work directory from `<program_root>/work` to `<user_home_dir>/.spinetoolbox/work`. Current user's only see this change if they go to Settings->General and clear the work directory line edit.

**Comment by soininen** [comment]
I'd rather have a single `.spinetoolbox` directory under user's home directory with subdirectories to `work` and `plugins` as I am not personally fond of multiple `.spinetoolbox*` directories cluttering my home. Further, I guess *NIX systems already have a config dir for Toolbox which we might be able to reuse. Perhaps @manuelma can comment on that?

**Comment by manuelma** [comment]
I like the concept but not the outcome of the bikeshedding. `.spinetoolboxdata` is a bit too long. For other applications, the dot folder in the home is named after the app, so in our case it should be just `.spinetoolbox`.

Is the home folder ever going to be a spinetoolbox project for this to become a source of conflict? We can check in the code if the user is selecting the home folder as a project folder and prevent it with a nice message?

**Comment by PekkaSavolainen** [comment]
Done. The plugins dir can stay where it is and I just changed the default work directory to `<home_dir>/.spinetoolbox/work` and update some file dialog initial directories (New project and Open project now opens in home dir by default). I just remembered that there's no default `projects` directory anywhere in our code. It's up to the user to create one. We just nudge new users towards the home directory in New project, and Open project file dialogs now.
